### DISCUSSION TITLE
Convert pdf to a fillable form?

### DISCUSSION THREAD

**Discussion by Rapid1898-code** [jsvine/pdfplumber#838] [Ask for help with specific PDFs]
I am not sure if this questions is fully related to pdfplumber - but least my question is about a pdf... :=)

I have a pdf-template and would like to convert this pdf to a fillable pdf-form.
(the file has a main header area and a table with different informations over several sites - as you can see in the attached file)
[input.pdf](https://github.com/jsvine/pdfplumber/files/10955013/input.pdf)

Is there any way how i can use python in some way to help here?
(i know xlwings, pdfplubmer, fillpdf which probably can help - but how can i convert the existing pdf to a pdf with fillable fields so i can use it with eg. fillpdf?)

**Comment by jsvine** [comment]
Hi @Rapid1898-code, you're correct that this isn't something `pdfplumber` can do; this library is focused on *reading* PDFs, not *writing* them. It's possible that it could be used, however, in combination with other tools (e.g., to get the coordinates of certain objects on the page). I'm not quite as familiar with the PDF-writing/editing tools, but you might try these options:

- https://pypdf.readthedocs.io/en/stable/
- https://pymupdf.readthedocs.io/en/latest/
